### DISCUSSION TITLE
Reading sqlite file which has multiple geometry columns in tables

### DISCUSSION THREAD

**Discussion by danielissac** [geopandas/geopandas#2843] [Q&A]
I'm new to Geopandas. When I try to read sqlite file which has multiple geometry columns in each tables, I'm getting only one geometry column per table in the resultant GeoDataFrame.

For example:
A table has geometry type columns like _bottom_left_, _bottom_right_, _top_left_ and _top_right_. When I read the file using `gpd.read_file(path_to_sqlite_file, engine="pyogrio")` method, I'm getting only one geometry column (column name is also renamed to geometry) in the Geodataframe. How do I read all the geometry type columns?

**Comment by brendan-ward** [comment]
Are the columns all named "geometry"?  If so, [this comment on #2752](https://github.com/geopandas/geopandas/issues/2752#issuecomment-1374645026) might be helpful; you just end up doing some of the things that pyogrio does internally.

If they are different column names, are they still added as fields but only one of them is decoded as a GeoSeries, or are they absent altogether?  It is expected that only one of them is decoded as a GeoSeries, but if they are present with WKB bytes data, you should be able to convert the convert the other fields once you've pulled them into GeoPandas.

If you are able to share a tiny example file that has the same structure, that may help us investigate this further.

**Comment by brendan-ward** [comment]
What does the following show you?

```Python
from pyogrio import read_info

print(read_info(path_to_sqlite_file))
```

**Reply by danielissac** [reply]
I'm getting this info.
`{'crs': 'EPSG:4326', 'encoding': 'cp1252', 'fields': array(['intersection_id', 'type'], dtype=object), 'dtypes': array(['int32', 'int32'], dtype=object), 'geometry_type': 'Point', 'features': 0, 'capabilities': {'random_read': 1, 'fast_set_next_by_index': 0, 'fast_spatial_filter': 0}}`

**Reply by danielissac** [reply]
for `read_info(path_to_sqlite_file,layer=table_name)`, I'm getting this info.

```
{'crs': 'EPSG:4326', 'encoding': 'cp1252', 'fields': array(['sign_id', 'sign_face_azimuth', 'sign_face_elevation', 'height',
       'width', 'shape', 'type', 'is_digital', 'sign_type_confidence',
       'sign_geometry_confidence', 'value'], dtype=object), 'dtypes': array(['int32', 'float64', 'float64', 'float64', 'float64', 'int32',
       'int32', 'bool', 'float64', 'float64', 'int32'], dtype=object), 'geometry_type': '2.5D Point', 'features': 2045, 'capabilities': {'random_read': 1, 'fast_set_next_by_index': 0, 'fast_spatial_filter': 0}}
```

**Comment by brendan-ward** [comment]
Ok, that clarifies.  Pyogrio is not seeing them because they are true geometry columns rather than non-geometry fields (e.g., blob fields holding WKB).

Related: https://github.com/geopandas/pyogrio/issues/165

Currently not supported, so you are going to need to read each geometry field individually and join them up in Python.  I don't have a file of this structure to test with, but I think you could create a view in your SQLite database for each geometry column and the primary key, then use that in place of `table_name` and join the tables back together on your primary key.

**Reply by danielissac** [reply]
Thanks for the help. I'll give a try.

**Reply by jorisvandenbossche** [reply]
It might be that the arrow-based reading already supports reading multiple geometry columns. Could you try with passing `use_arrow=True` in the `gpd.read_file(..., engine="pyogrio", use_arrow=True)` call?

That will require to install pyarrow.

**Reply by danielissac** [reply]
Hey, Thanks a lot for the answer. Finally, I'm able to read all the geometry type columns by passing `use_arrow = True` but in the resultant GeoDataFrame geometry type columns are converted to WKB representation. By using [from_wkb](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.from_wkb.html) method, I'm able to convert it back to geometry type. Thanks for the help @jorisvandenbossche 🥂
